ui_framework/components/accessibility/keyboard_accessible.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Interactive elements must be able to receive focus.
+ *
+ * Ideally, this means using elements that are natively keyboard accessible (<a href="">,
+ * <input type="button">, or <button>). Note that links should be used when navigating and buttons
+ * should be used when performing an action on the page.
+ *
+ * If you need to use a <div>, <p>, or <a> without the href attribute, then you need to allow
+ * them to receive focus and to respond to keyboard input. The workaround is to:
+ *
+ *   - Give the element tabindex="0" so that it can receive keyboard focus.
+ *   - Add a JavaScript onkeyup event handler that triggers element functionality if the Enter key
+ *     is pressed while the element is focused. This is necessary because some browsers do not trigger
+ *    onclick events for such elements when activated via the keyboard.
+ *   - If the item is meant to function as a button, the onkeyup event handler should also detect the
+ *     Spacebar in addition to the Enter key, and the element should be given role="button".
+ *
+ * Wrap any of these elements in this component to automatically do the above.
+ */
+
+import {
+  Component,
+  cloneElement,
+} from 'react';
+
+import {
+  accessibleClickKeys,
+  SPACE_KEY,
+} from '../../services';
+
+export class KuiKeyboardAccessible extends Component {
+  onKeyDown = e => {
+    // If the user is interacting with a different element, then we don't need to do anything.
+    if (e.currentTarget !== e.target) {
+      return;
+    }
+
+    // Prevent a scroll from occurring if the user has hit space.
+    if (e.keyCode === SPACE_KEY) {
+      e.preventDefault();
+    }
+  }
+
+  onKeyUp = e => {
+    // If the user is interacting with a different element, then we don't need to do anything.
+    if (e.currentTarget !== e.target) {
+      return;
+    }
+
+    // Support keyboard accessibility by emulating mouse click on ENTER or SPACE keypress.
+    if (accessibleClickKeys[e.keyCode]) {
```

Comment:
nitpick. I think I'd prefer `e.keyCode === ENTER_KEY || e.keyCode === SPACE_KEY`, just to be clear. Or maybe make it a function instead `isAccessibleClickKey(e.keyCode)`? (I just think it reads better)

In either case I think the comment can be removed.